### PR TITLE
Support groups in `TransformMatchingParts`

### PR DESCRIPTION
**Pull request by NicoWeio**
## Motivation and Explanation: Why and how do your changes improve the library?
I was looking for an elegant way to update an equation, letting some variables “fly in” from elsewhere.
`TransformMatchingTex` is great, but it only supported one `MathTex`-like input and one output.

This PR adds support for `Group`s (and `VGroup`s) by accessing their individual `submobject`s. 
I made it recursive, so that nested groups are supported.
Mobjects that are neither groups, nor contain a `tex_string`, will still raise an Exception.

## Links to added or changed documentation pages

https://manimce--2602.org.readthedocs.build/en/2602/reference/manim.animation.transform_matching_parts.TransformMatchingTex.html

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by NicoWeio** [comment]
I wonder if the example in 6c6cd90 is good enough. One might consider integrating the previous example (without groups) or creating a second one.
In fact, I already did No. 1:

```python
class MatchingEquationParts(Scene):
    def construct(self):
        variables = VGroup(MathTex("a"), MathTex("b"), MathTex("c")).arrange_submobjects().shift(UP)

        eq1 = MathTex("{{x}}^2", "+", "{{y}}^2", "=", "{{z}}^2")
        eq2 = MathTex("{{a}}^2", "+", "{{b}}^2", "=", "{{c}}^2")
        eq3 = MathTex("{{a}}^2", "=", "{{c}}^2", "-", "{{b}}^2")

        self.add(eq1)
        self.wait(0.5)
        self.play(TransformMatchingTex(Group(eq1, variables), eq2))
        self.wait(0.5)
        self.play(TransformMatchingTex(eq2, eq3))
        self.wait(0.5)
```

**Comment by Kiran-Raj-Dev** [comment]
I think having some form of splitting is nice then no control.
This is certainly a change for the better!

File: manim/animation/transform_matching_parts.py [hunk 1]
```diff
@@ -276,7 +277,15 @@ def __init__(
 
     @staticmethod
     def get_mobject_parts(mobject: Mobject) -> list[Mobject]:
-        return mobject.submobjects
+        if isinstance(mobject, (Group, VGroup)):
```
Comment (behackl) [inline manim/animation/transform_matching_parts.py:280]: mobject should also be allowed to be an `OpenGLGroup` or an `OpenGLVGroup`.

**Review by behackl — CHANGES_REQUESTED** [review]
Thank you for your contribution, this looks good and almost ready to merge to me!

I've left a request for a minor change, please take a look.

**Review by behackl — APPROVED** [review]
Thank you!

**Pull request merged**